src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+using TestEnvironment.Docker.ContainerLifecycle;
+
+namespace TestEnvironment.Docker.Containers.Redis
+{
+    public class RedisContainerWaiter : BaseContainerWaiter<RedisContainer>
+    {
+        public RedisContainerWaiter()
+        {
+        }
+
+        public RedisContainerWaiter(ILogger logger)
+            : base(logger)
+        {
+        }
+
+        protected override async Task<bool> PerformCheckAsync(RedisContainer container, CancellationToken cancellationToken)
+        {
+            RedisConnectionConfiguration redisConnectionConfiguration = container.GetConnectionConfiguration();
+
+            var redisConfigurationOptions = new ConfigurationOptions()
+            {
+                EndPoints =
+                {
+                    { redisConnectionConfiguration.Host, redisConnectionConfiguration.Port },
+                },
+                Password = redisConnectionConfiguration.Password
+            };
+
+            var redis = await ConnectionMultiplexer.ConnectAsync(redisConfigurationOptions);
+
+            var s = redis.GetStatus();
```

Comment:
Do you need to check the status here or it is really meaningless and just successful response is enough?

Reply:
If there is connection/password issue, connectAsync will throw exception. Once the connectAsync passed it is safe to assume that the redis is up and running. In addition to that GetStatus will also give successful response if connection is successful.

This option we have is safe.

Reply:
In that case you can replace `s` variable with `_`.

Reply:
Introduced the discard variable.